docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,446 @@
+.. _building_the_project_with_dpcpp:
+
+Building the Project with DPC++
+===============================
+
+This page describes building the oneMKL interface library with either the Intel(R) oneAPI DPC++ Compiler or open-source oneAPI DPC++ Compiler.
+For guidance on building the project with AdaptiveCpp, see :ref:`building_the_project_with_adaptivecpp`.
+
+.. _build_setup_with_dpcpp:
+
+Build Setup
+###########
+
+#. 
+   Install Intel(R) oneAPI DPC++ Compiler (select the variant as described in
+   :ref:`Selecting a Compiler<selecting_a_compiler>`).
+
+#. 
+   Clone this project to ``<path to onemkl>``\ , where ``<path to onemkl>``
+   is the root directory of this repository.
+
+#. 
+   Build and install all `required dependencies <https://github.com/oneapi-src/oneMKL?tab=readme-ov-file#software-requirements>`_. 
+
+.. _build_introduction_with_dpcpp:
+
+Introduction
+############
+
+In most cases, building oneMKL is as simple as setting the compiler and selecting the desired backends to build with.
```

Comment:
Could you please clarify the intention of this sentence? It gave me an out-of-nowhere feeling.

Reply:
It has two purposes:
* It gives a preamble before the example cmake invocation. Starting with a cmake invocation would seem sudden to me.
* To try and make building oneMKL relatively simple.
  * All of the details mean that these build instructions are compilicated. We don't want a potential user to be discouraged. 
  * Most users won't have to look at most of the instructions (I hope). I'd like to encourage them to try building as step one, then fix any backend-specific issues.

Reply:
I see. Maybe we can expand a little and give some orientation, like:
>The build commands for various compilers and backends differ mostly in setting the
values of CMake options for compiler and backend. In this section, we describe the
common build commands. We will discuss backend-specific details in the \`Backends\`_
section and provide examples in \`CMake invocation examples\`_.

Feel free to edit it.

Reply:
Your version is better. Added in [04744c6](https://github.com/oneapi-src/oneMKL/pull/510/commits/04744c67f87fb9b9be5202cdc1cdc06078a1d59e).